src/language-js/printer-estree.js

```diff
@@ -1631,12 +1631,19 @@ function printPathNoParens(path, options, print, args) {
       return concat(["debugger", semi]);
     case "JSXAttribute":
       parts.push(path.call(print, "name"));
-
       if (n.value) {
         let res;
         if (isStringLiteral(n.value)) {
+          let regex = /"/g;
+          let quote = '"';
+          let escapedChar = "&quot;";
+          if (options.jsxSingleQuote) {
+            regex = /'/g;
+            escapedChar = "&apos;";
+            quote = "'";
+          }
           const value = rawText(n.value);
-          res = '"' + value.slice(1, -1).replace(/"/g, "&quot;") + '"';
+          res = quote + value.slice(1, -1).replace(regex, escapedChar) + quote;
```

Comment:
Can you add something here to replace the `&apos;` in `'foo &apos; bar'` with `'` when `jsxSingleQuote` is false?

Reply:
Hey j-f1, I can do that. In a similar manner, would we also want to replace `&quot;` in  `"foo &quot; bar"` with `"` when `jsxSingleQuote` is true?

Reply:
That would be great!